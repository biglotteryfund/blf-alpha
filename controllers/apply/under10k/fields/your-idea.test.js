/* eslint-env jest */
'use strict';
const { fieldYourIdeaPriorities } = require('./your-idea');

test('show COVID-19 guidance text in England', function () {
    const data = {
        projectCountry: 'england',
    };
    const field = fieldYourIdeaPriorities('en', data);
    expect(field.explanation).toMatchSnapshot();
});

test('show COVID-19 guidance text outside England when project is responding to COVID-19', function () {
    const data = {
        projectCountry: 'scotland',
        supportingCOVID19: 'yes',
    };
    const field = fieldYourIdeaPriorities('en', data);
    expect(field.explanation).toMatchSnapshot();
});

test('show regular guidance text outside England when project is not responding to COVID-19', function () {
    const data = {
        projectCountry: 'scotland',
        supportingCOVID19: 'no',
    };
    const field = fieldYourIdeaPriorities('en', data);
    expect(field.explanation).toMatchSnapshot();
});

test('show combined guidance text as a fallback', function () {
    const field = fieldYourIdeaPriorities(
        'en',
        {}
    );
    expect(field.explanation).toMatchSnapshot();
});
