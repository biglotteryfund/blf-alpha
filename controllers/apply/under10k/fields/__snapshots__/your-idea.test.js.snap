// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show COVID-19 guidance text in England 1`] = `
<p>
  We'll prioritise organisations supporting people and communities who experience 
disproportionate challenge and difficulty as a result of the COVID-19 crisis. This category
includes groups which are facing specific challenges during the current crisis and includes
organisations supporting:
</p>
<ul>
  <li>
    older people
  </li>
  <li>
    disabled people (including people with long-term health conditions)
  </li>
  <li>
    lesbian, gay, bisexual, transgender, queer + (LGBTQ+) communities
  </li>
  <li>
    Black, Asian and Minority Ethnic (BAME) communities.
  </li>
</ul>
<p>
  It also includes organisations who provide advice and support to those pushed into crisis as a 
result of the pandemic, and those organisations supporting people and their families with 
end-of-life care.
</p>
<p>
  <strong>
    You can write up to 150 words for this section,
but don't worry if you use less.
  </strong>
</p>
`;

exports[`show COVID-19 guidance text in England 2`] = `
<p>
  We'll prioritise organisations supporting people and communities who experience 
disproportionate challenge and difficulty as a result of the COVID-19 crisis. This category
includes groups which are facing specific challenges during the current crisis and includes
organisations supporting:
</p>
<ul>
  <li>
    older people
  </li>
  <li>
    disabled people (including people with long-term health conditions)
  </li>
  <li>
    lesbian, gay, bisexual, transgender, queer + (LGBTQ+) communities
  </li>
  <li>
    Black, Asian and Minority Ethnic (BAME) communities.
  </li>
</ul>
<p>
  It also includes organisations who provide advice and support to those pushed into crisis as a 
result of the pandemic, and those organisations supporting people and their families with 
end-of-life care.
</p>
<p>
  <strong>
    You can write up to 150 words for this section,
but don't worry if you use less.
  </strong>
</p>
`;

exports[`show COVID-19 guidance text outside England when project is responding to COVID-19 1`] = `
<p>
  <strong>
    We will prioritise:
  </strong>
</p>
<ol>
  <li>
    organisations supporting people and communities
who experience disproportionate challenge and
difficulty as a result of the COVID-19 crisis
  </li>
  <li>
    organisations providing services and support
for vulnerable people, for which there will be
increased demand as a result of the COVID-19 crisis
  </li>
  <li>
    organisations which connect communities and support
communities to work together to respond to COVID-19.
  </li>
</ol>
<p>
  You can tell us if your project meets more
than one priority, but don't worry if it doesn't.
</p>
<p>
  <strong>
    You can write up to 150 words for this section,
but don't worry if you use less.
  </strong>
</p>
`;

exports[`show combined guidance text as a fallback 1`] = `
<p>
  <strong>
    We will prioritise:
  </strong>
</p>
<ol>
  <li>
    organisations supporting people and communities
who experience disproportionate challenge and
difficulty as a result of the COVID-19 crisis
  </li>
  <li>
    organisations providing services and support
for vulnerable people, for which there will be
increased demand as a result of the COVID-19 crisis
  </li>
  <li>
    organisations which connect communities and support
communities to work together to respond to COVID-19.
  </li>
</ol>
<p>
  You can tell us if your project meets more
than one priority, but don't worry if it doesn't.
</p>
<p>
  <strong>
    You can write up to 150 words for this section,
but don't worry if you use less.
  </strong>
</p>
`;

exports[`show regular guidance text outside England when project is not responding to COVID-19 1`] = `
<p>
  <strong>
    We will prioritise:
  </strong>
</p>
<ol>
  <li>
    organisations supporting people and communities
who experience disproportionate challenge and
difficulty as a result of the COVID-19 crisis
  </li>
  <li>
    organisations providing services and support
for vulnerable people, for which there will be
increased demand as a result of the COVID-19 crisis
  </li>
  <li>
    organisations which connect communities and support
communities to work together to respond to COVID-19.
  </li>
</ol>
<p>
  You can tell us if your project meets more
than one priority, but don't worry if it doesn't.
</p>
<p>
  <strong>
    You can write up to 150 words for this section,
but don't worry if you use less.
  </strong>
</p>
`;
